src/core/index.ts

```diff
@@ -239,13 +246,13 @@ export class ZarrArray {
 
   public get(selection?: undefined | Slice | ":" | "..." | null | (Slice | null | ":" | "...")[]): Promise<NestedArray<TypedArray>>;
   public get(selection?: ArraySelection): Promise<NestedArray<TypedArray> | number>;
-  public get(selection: ArraySelection = null): Promise<NestedArray<TypedArray> | number> {
-    return this.getBasicSelection(selection);
+  public get(selection: ArraySelection = null, options?: StoreGetOptions): Promise<NestedArray<TypedArray> | number> {
```

Comment:
You can simplify the code for the defaults in the options object by using destructuring in the function signature.

[Example usage in **`creation.ts`**](https://github.com/gzuidhof/zarr.js/blob/master/src/creation.ts#L73)

Here it would be 
```typescript
private async getSelection(indexer: BasicIndexer, {concurrencyLimit = 10}: StoreGetOptions)
```
In this case the callback will be `undefined` is not passed :)

Then in the other functions that are called from here, I would make it a non-optional argument.

Reply:
I'll have a go at this. I think I understand now. Thanks for the guidance